x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/job_map.tsx

```diff
@@ -151,20 +127,63 @@ export const JobMap: FC<Props> = ({ analyticsId, modelId }) => {
     <>
       <EuiSpacer size="m" />
       <div style={{ height: height - parseInt(theme.gutterTypes.gutterLarge, 10) }} ref={ref}>
-        <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexGroup direction="column" gutterSize="none" justifyContent="spaceBetween">
           <EuiFlexItem grow={false}>
-            <JobMapTitle analyticsId={analyticsId} modelId={modelId} />
+            <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
+              <EuiFlexItem grow={false}>
+                <JobMapTitle analyticsId={analyticsId} modelId={modelId} />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <JobMapLegend />
+              </EuiFlexItem>
+            </EuiFlexGroup>
           </EuiFlexItem>
           <EuiFlexItem grow={false}>
-            <JobMapLegend />
+            <EuiFlexGroup gutterSize="xs" component="span">
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  size="xs"
+                  data-test-subj={`mlAnalyticsRefreshMapButton${
+                    isLoading ? ' loading' : ' loaded'
+                  }`}
+                  onClick={() => getDataWrapper({ analyticsId, modelId })}
+                  isLoading={isLoading}
+                >
+                  <FormattedMessage
+                    id="xpack.ml.dataframe.analyticsList.refreshMapButtonLabel"
+                    defaultMessage="Refresh"
+                  />
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  size="xs"
+                  data-test-subj="mlAnalyticsResetGraphButton"
+                  onClick={() => setResetCy(!resetCy)}
+                >
+                  <FormattedMessage
+                    id="xpack.ml.dataframe.analyticsList.resetMapButtonLabel"
+                    defaultMessage="Reset"
+                  />
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+            </EuiFlexGroup>
           </EuiFlexItem>
         </EuiFlexGroup>
-        <Cytoscape height={height} elements={elements} width={width} style={cytoscapeDivStyle}>
+        <Cytoscape
+          height={height}
```

Comment:
And to avoid the inner scrollbar on the map, this works on my setup (subtracting a further 50px for the height of the title and reset / refresh button elements):

```
height - parseInt(theme.gutterTypes.gutterLarge, 10) - 70
```